defaults/main.yml

```diff
@@ -41,16 +41,67 @@ docker__distribution_release: '{{ ansible_local.core.distribution_release
 docker__upstream: False
 
                                                                    # ]]]
+# .. envvar:: docker__variant [[[
+#
+# For upstream repositories the variant to be installed: ce or ee. Note that Docker EE
+# is not supported on Debian.
+docker__variant: 'ce'
+
+                                                                   # ]]]
+# .. envvar:: docker__channel [[[
+#
+# For upstream repositories choose the stable or edge channel.
+docker__channel: 'stable'
+
+                                                                   # ]]]
 # .. envvar:: docker__upstream_key [[[
 #
 # APT GPG key id used to sign the upstream Docker packages.
-docker__upstream_key: '58118E89F3A912897C070ADBF76221572C52609D'
+docker__upstream_key: '{{ "9DC858229FC7DD38854AE2D88D81803C0EBFCD88"
+                           if (docker__variant == "ce")
+                           else "DD911E995A64A202E85907D6BC14F10B6D085F96" }}'
+
+                                                                   # ]]]
+# .. envvar:: docker__version [[[
+#
+# The docker version to be installed. Leave empty to install the latest (stable or edge)
+# version, or specify a version. Example:
+#
+# .. code-block:: yaml
+#    :linenos:
+#
+#    docker__version: '17.05.0~ce-0~ubuntu-xenial'
+#
+docker__version: ''
```

Comment:
As I understand it you pick the specific version to be installed by APT. But doing this directly instead of via APT preference pins will result in the package being upgraded when the next version is released. If you would want to allow selecting a specific version, doing this via APT preferences, and perhaps using `debops.apt_preferences`, would be a better way.

Since this would most likely only apply to the upstream packages, I would rename this variable to `docker__upstream_version`. The distribution usually stick to one Docker version, so there's no need to specify it directly.